File: solutions/typescript/prime-factors/1/prime-factors.ts
export function calculatePrimeFactors(value: number) {
    const results: number[] = [];
    let n = value;

    while (n % 2 === 0) {
        results.push(2);
        n /= 2;
    }

    let d = 3;
    while (d * d <= n) {
        while (n % d === 0) {
            results.push(d);
            n /= d;
        }
        d += 2;
    }

    if (n > 1) {
        results.push(n);
    }

    return results;
}

File: solutions/go/prime-factors/1/prime_factors.go
package primefactors

func Factors(n int64) []int64 {
	results := []int64{}
	num := n
	for num%2 == 0 {
		results = append(results, 2)
		num /= 2
	}

	d := int64(3)
	for d*d <= n {
		for num%d == 0 {
			results = append(results, d)
			num /= d
		}
		d += 2
	}

	if num > 1 {
		results = append(results, num)
	}
	return results
}
